lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -96,7 +99,8 @@ def transform_block_filters(self, usage_info, block_structure):
                 selected = block_keys['selected']
 
                 # Save back any changes
-                if any(block_keys[changed] for changed in ('invalid', 'overlimit', 'added')):
+                if usage_info.user.is_authenticated() and any(block_keys[changed]
+                                                              for changed in ('invalid', 'overlimit', 'added')):
```

Comment:
How about refactoring the two line one-liner? Something like:
```
block_keys_changed = any(block_keys[changed] for changed in ('invalid', 'overlimit', 'added'))
if usage_info.user.is_authenticated() and block_keys_changed:
```

Reply:
This is from an earlier PR and was based on the review comments. Once that is merged, it won't show up here.